script/makeinstaller.sh

```diff
@@ -1,62 +1,81 @@
 #!/usr/bin/env bash
-# Execute this at the base of the streamlink repo.
+set -e
+
+MAKEINSTALLER=$(basename "$(readlink -f "${0}")")
+log() {
+    echo "[${MAKEINSTALLER}] $@"
+}
+err() {
+    log >&2 "$@"
+    exit 1
+}
+
+
+declare -A DEPS=(
+    [makensis]=NSIS
+    [pynsist]=pynsist
+    [convert]=Imagemagick
+    [inkscape]=inkscape
+    [curl]=curl
+    [jq]=jq
+    [unzip]=unzip
+)
+
+for dep in "${!DEPS[@]}"; do
+    command -v "${dep}" 2>&1 >/dev/null || err "${DEPS["${dep}"]} is required to build the installer. Aborting."
+done
 
-set -e # stop on error
+ROOT=$(git rev-parse --show-toplevel 2>/dev/null || realpath "$(dirname "$(readlink -f "${0}")")/..")
+cd "${ROOT}"
 
-command -v makensis > /dev/null 2>&1 || { echo >&2 "makensis is required to build the installer. Aborting."; exit 1; }
-command -v pynsist > /dev/null 2>&1 || { echo >&2 "pynsist is required to build the installer. Aborting."; exit 1; }
-command -v convert > /dev/null 2>&1 || { echo >&2 "imagemagick is required to build the installer. Aborting."; exit 1; }
-command -v inkscape > /dev/null 2>&1 || { echo >&2 "inkscape is required to build the installer. Aborting."; exit 1; }
 
 # For CI nightly builds generate a version number with commit hash
 STREAMLINK_VERSION=$(python setup.py --version)
 STREAMLINK_VERSION_PLAIN="${STREAMLINK_VERSION%%+*}"
 STREAMLINK_INSTALLER="${1:-"streamlink-${STREAMLINK_VERSION/\+/_}"}"
+STREAMLINK_PYTHON_VERSION=3.6.6
+STREAMLINK_ASSETS_REPO="streamlink/streamlink-assets"
+STREAMLINK_ASSETS_RELEASE="latest"
 
-# include the build number
 CI_BUILD_NUMBER=${GITHUB_RUN_ID:-0}
 STREAMLINK_VI_VERSION="${STREAMLINK_VERSION_PLAIN}.${CI_BUILD_NUMBER}"
 
-build_dir="$(pwd)/build"
+DIST_DIR="${STREAMLINK_DIST_DIR:-"${ROOT}/dist"}"
+INSTALLER_PATH="${DIST_DIR}/${STREAMLINK_INSTALLER}.exe"
+
+build_dir="${ROOT}/build"
 build_dir_plugins="${build_dir}/lib/streamlink/plugins"
+cache_dir="${build_dir}/cache"
 nsis_dir="${build_dir}/nsis"
 files_dir="${build_dir}/files"
 icons_dir="${files_dir}/icons"
-# get the dist directory from an environment variable, but default to the build/nsis directory
-dist_dir="${STREAMLINK_DIST_DIR:-${nsis_dir}}"
-mkdir -p "${build_dir}" "${dist_dir}" "${nsis_dir}" "${files_dir}" "${icons_dir}"
 
-echo "Building streamlink-${STREAMLINK_VERSION} package..." 1>&2
+removed_plugins_file="${ROOT}/src/streamlink/plugins/_removed.txt"
```

Comment:
I personally prefer `.removed.txt` as the name for that file, or even just `.removed`

Reply:
The idea was to have the file sorted at the top. A leading dot would hide it in most *nix file managers by default.

Reply:
Hidden is the best place for it, imo.

Reply:
Sure, let's change it. Will update after some more feedback and issue resolving.